chips/nrf52/src/chip.rs

```diff
@@ -26,7 +24,142 @@ impl<I: InterruptService> NRF52<I> {
     }
 }
 
-impl<I: InterruptService> kernel::Chip for NRF52<I> {
+/// This macro defines a struct that, when initialized,
+/// instantiates all peripheral drivers shared across all supported nrf52 chips
+/// in Tock. If a board
+/// wishes to use only a subset of these peripherals, this
+/// macro cannot be used, and this struct should be
+/// constructed manually in main.rs. The input to the macro is the name of the struct
+/// that will hold the peripherals, which can be chosen by the board.
+#[macro_export]
+macro_rules! create_default_nrf52_peripherals {
```

Comment:
I'm not sure that "create" is the clearest name for this macro. Without context, I had assumed that the contents of the macro were some code that initializes the peripherals, to put inside a function (similarly to `static_init`). I didn't assume the macro's contents were a struct definition.

Can we come up with a better naming?

Reply:
How about `define_default_nrf52_peripherals`?